add dry option

tpo opened the pull request:
Hi,

this pull request should implement #4 - the `dry` option. It also improves a few other things as error messages and logs more things.

I have *not* added tests for the `dry` option, nor have I verified whether the modified error messages and the additional logs pass the tests. I'd be nice to have a note in the documentation on how these tests are supposed to be run, then I'd could have a look:

```
bash-5.1# rake test
/usr/local/bin/ruby -w -I"lib:lib" /usr/local/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/rake_test_loader.rb "test/failover_ip_test.rb" "test/hooks_test.rb" 

File does not exist: test/unit

rake aborted!
Command failed with status (1): [ruby -w -I"lib:lib" /usr/local/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/rake_test_loader.rb "test/failover_ip_test.rb" "test/hooks_test.rb" ]

Tasks: TOP => test
(See full trace by running task with --trace)
```

OT: I have tried to understand the failover_ip/ping_ip/ips: ping/target semantics by re-re-re-re-rereading the README and after that by studying the code, but I have not succeeded so far... a clearer README wrt to those semantics would be gold.

mrkamel commented:
hi, thanks for your efforts. the project is a bit out of date, last update from 2015. have to bring it up to date first. created a PR to migrate to minitest. https://github.com/mrkamel/heartbeat/pull/11 Using that you should be able you to run the tests now.

tpo commented:
@mrkamel thanks a lot for the minitest bits.

I've fixed stuff so that the tests run through plus i've added `dry: true` tests.

I have two branches in my fork:

* [the master branch](https://github.com/sourcepole/heartbeat) is based on your master (that is without minitest)
* [in the merge_master_with_upstream_use_minitest branch](https://github.com/sourcepole/heartbeat/tree/merge_master_with_upstream_use_minitest) I have merged my master with your minitest branch and fixed stuff and added tests as per above

In case you want to include my changes into your tree - which I certainly would very much welcome - then I suggest that:

1. you merge your minitest branch into your master (which should be conflict free and therefore a fastforward (non-)merge)
2. I then reapply all my patches on top of your master

If I don't do 2. then I'm afraid you will have conflicts when merging either of my branches into either of your branches, since:

- you applied the minitest patches on top of your master branch
- I've been working in a fork of your master branch pre-minitest and once tests ran through in the merge_master_with_upstream_use_minitest branch I've cherry picked changes from it 
- my merge_master_with_upstream_use_minitest is originaly a branch off of my master with your minitest branch merged (with resolved conflicts!) and then some more patches on top of it all

So pulling/merging will not be very pretty for you. Maybe you don't care and then you can merge right away. There should be only a few conflicts, but better try it out first.

If you want to have a clean tree then let me know and I'll do 2. above.

tpo commented:
> hi, could you maybe rebase this one then or create a new PR using the rebased version? thx alot

I will, however I think #13 should be resolved before I rebase - agreed?

mrkamel commented:
> I will, however I think #13 should be resolved before I rebase - agreed?

yup, fixed

tpo commented:
A hi, could you maybe rebase this one then or create a new PR using the rebased version? thx alot

closed in favor of #15 that is based on the rebased branch of mine
